s3api/handlers_buckets.go

```diff
@@ -218,25 +223,67 @@ func (s *server) BucketHeadHandler(w http.ResponseWriter, r *http.Request) {
 	w.Write([]byte{})
 }
 
-// BucketDeleteHandler deletes an empty bucket
+// BucketDeleteHandler deletes an empty bucket and all of it's dependencies.  The operations are
+// 1. the bucket is deleted, this will fail if the bucket is not empty
+// 2. a list of policies attached to the bucket admin group (<bucketName>-BktAdmGrp) is gathered
+// 3. each of those policies is detached from the group and if the ut starts with '<bucketName>-' it is deleted
+// 4. the bucket admin group is deleted
 func (s *server) BucketDeleteHandler(w http.ResponseWriter, r *http.Request) {
 	w = LogWriter{w}
 	vars := mux.Vars(r)
 	account := vars["account"]
 	bucket := vars["bucket"]
+
 	s3Service, ok := s.s3Services[account]
 	if !ok {
 		log.Errorf("account not found: %s", account)
 		w.WriteHeader(http.StatusNotFound)
 		return
 	}
 
+	iamService, ok := s.iamServices[account]
+	if !ok {
+		msg := fmt.Sprintf("IAM service not found for account: %s", account)
+		handleError(w, apierror.New(apierror.ErrNotFound, msg, nil))
+		return
+	}
+
 	output, err := s3Service.DeleteEmptyBucket(r.Context(), &s3.DeleteBucketInput{Bucket: aws.String(bucket)})
 	if err != nil {
 		handleError(w, err)
 		return
 	}
 
+	groupName := fmt.Sprintf("%s-BktAdmGrp", bucket)
+	policies, err := iamService.ListGroupPolicies(r.Context(), &iam.ListAttachedGroupPoliciesInput{GroupName: aws.String(groupName)})
+	if err != nil {
+		handleError(w, err)
```

Comment:
I'm not sure what's the best way to do this, but in this case if the `ListGroupPolicies` fails for whatever reason the s3-api will return an error. You can argue that if the bucket was deleted this was a success. Subsequent DELETE calls will fail because the bucket doesn't exist and there's no way to clean up those groups/policies anyway.
Maybe for these additional steps we can still return a `200` but also append some warning message?

Reply:
we can do that - but keep in mind that trying to recreate a group or policy with the same name will fail, so if we fail to cleanup, you wont be able to create a new bucket with that same name, so we at least need to notify in slack

Reply:
yeah, we can notify and manually clean up, it should be an edge case